File: frontend/src/services/orcamentoService.ts
import api from './api';
import { Orcamento, OrcamentoProgress } from '../types/Orcamento';

export const orcamentoService = {
    listarTodos: async (): Promise<Orcamento[]> => {
        const response = await api.get('/orcamentos');
        return response.data;
    },

    buscarPorId: async (id: number): Promise<Orcamento> => {
        const response = await api.get(`/orcamentos/${id}`);
        return response.data;
    },

    buscarPorPeriodo: async (mes: number, ano: number): Promise<Orcamento[]> => {
        const response = await api.get(`/orcamentos/periodo?mes=${mes}&ano=${ano}`);
        return response.data;
    },

    getProgresso: async (mes: number, ano: number): Promise<OrcamentoProgress[]> => {
        const response = await api.get(`/orcamentos/progresso?mes=${mes}&ano=${ano}`);
        return response.data;
    },

    criar: async (data: Omit<Orcamento, 'id'>): Promise<Orcamento> => {
        const response = await api.post('/orcamentos', data);
        return response.data;
    },

    atualizar: async (id: number, data: Partial<Orcamento>): Promise<Orcamento> => {
        const response = await api.put(`/orcamentos/${id}`, data);
        return response.data;
    },

    deletar: async (id: number): Promise<void> => {
        await api.delete(`/orcamentos/${id}`);
    },
};

File: backend/src/main/java/com/financeiro/service/OrcamentoService.java
package com.financeiro.service;

import com.financeiro.dto.OrcamentoProgressDTO;
import com.financeiro.dto.OrcamentoRequestDTO;
import com.financeiro.dto.OrcamentoResponseDTO;

import java.util.List;

public interface OrcamentoService {

    List<OrcamentoResponseDTO> listarTodos();

    OrcamentoResponseDTO buscarPorId(Long id);

    OrcamentoResponseDTO criar(OrcamentoRequestDTO dto);

    OrcamentoResponseDTO atualizar(Long id, OrcamentoRequestDTO dto);

    void deletar(Long id);

    List<OrcamentoResponseDTO> buscarPorPeriodo(Integer mes, Integer ano);

    List<OrcamentoProgressDTO> calcularProgresso(Integer mes, Integer ano);

    List<OrcamentoProgressDTO> calcularProgressoFamilia(String familia, Integer mes, Integer ano);
}
